src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -1167,16 +1166,27 @@ private String syncUserData(String userId) {
 		return errMsg;
 	}
 
-	public List<Map<String, Object>> getUserSearchData(String searchTerm) throws Exception {
+	public List<Map<String, Object>> getUserSearchData(String searchTerm, String rootOrgId) throws Exception {
 		List<Map<String, Object>> resultArray = new ArrayList<>();
 		Map<String, Object> result;
 		final BoolQueryBuilder query = QueryBuilders.boolQuery();
-		for (String field : serverConfig.getEsAutoCompleteSearchFields()) {
-			query.should(QueryBuilders.matchPhrasePrefixQuery(field, searchTerm));
-		}
-
 		final BoolQueryBuilder finalQuery = QueryBuilders.boolQuery();
-		finalQuery.must(QueryBuilders.termQuery(Constants.STATUS, 1)).must(query);
+		if(StringUtils.isNotEmpty(rootOrgId)){
+			searchTerm = searchTerm.toLowerCase();
+			MatchQueryBuilder matchQueryRootOrgId = QueryBuilders.matchQuery(Constants.ROOT_ORG_ID_RAW, rootOrgId);
+			for (String field : serverConfig.getEsAutoCompleteSearchFields()) {
+				query.should(new WildcardQueryBuilder(field, "*" + searchTerm + "*"));
+			}
+			TermQueryBuilder termQueryStatus = QueryBuilders.termQuery(Constants.STATUS_RAW, 1);
+			finalQuery.must(matchQueryRootOrgId);
+			finalQuery.must(query);
+			finalQuery.must(termQueryStatus);
+		} else{
+			for (String field : serverConfig.getEsAutoCompleteSearchFields()) {
+				query.should(QueryBuilders.matchPhrasePrefixQuery(field, searchTerm));
+			}
+			finalQuery.must(QueryBuilders.termQuery(Constants.STATUS, 1)).must(query);
```

Comment:
don't we change this too ??

Reply:
This "status" has been changed to "status.raw"